engine/schema/src/main/java/com/cloud/service/ServiceOfferingVO.java

```diff
@@ -327,7 +327,7 @@ public void setDynamicFlag(boolean isdynamic) {
     }
 
     public boolean isCustomCpuSpeedSupported() {
-        return isCustomized() && getDetail("minCPU") != null;
+        return isCustomized() && getDetail("minCPU") == null;
```

Comment:
it seems strange that the minimun cpu speed *must* be null and also that there are no conditions on the maximum cpu speed. Do you why this works @weizhouapache ?

Reply:
@DaanHoogland 
as suggested by @shwstppr , use `speed == null` instead, now it looks more clear

Reply:
@DaanHoogland 
there are two types of dynamic service offering
(1) constraint: speed is fixed. there are max/min cpu and memory.
(2) uncontraint: cpu/speed/memory are all dynamic.

the `isCustomCpuSpeedSupported` should be true, if the offering is unconstraint (speed is not set).
`getDetail("minCPU") != null` is a way to determine if the offering is uncontraint. `speed == null` is a better and simple way.